### DISCUSSION TITLE
Allowing custom attributes in custom HtmlPurifier config

### DISCUSSION THREAD

**Discussion by jr-cobweb** [craftcms/cms#8791] [Ideas]
### Description

It seems it's not possible to allow custom HTML attributes via custom HtmlPurifier config.

### Steps to reproduce

1. Create a new config and add a rule such as the following:

`
"HTML.AllowedAttributes": [
    "span.data-yyy",
    "span.data-xxx"
]
`

2. Attempt to save a document with one of the custom attributes, at which point an error will be thrown as purifier does not understand what the custom attributes are. I've had a quick look around and it seems you need to define custom attributes upfront, using the HtmlDefinition interface, which looks similar to the following:
`
$def->addAttribute('span', 'data-xxx', 'Number');
`
As far as I can tell there's no way to replicate the above interaction with the Purifier API via the config file.

### Additional info

- Craft version: 3.x
- PHP version: 7.2

**Comment by KarelJanVanHaute** [comment]
Does this have a solution yet? 
This is not working
``` "HTML.AllowedAttributes": ["data-*", "class", "id", "src", "href"],```
